test/integration/targets/zabbix_proxy/tasks/main.yml

```diff
@@ -0,0 +1,285 @@
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        proxy_name: zbxproxy_example01
+        description: Example Zabbix Proxy
+        state: present
+        status: active
+        proxy_address: 10.1.1.0/24,zabbix.example.com
+      register: zbxproxy_active_proxyaddress
+
+    - name: assert that proxy was updated
+      assert:
+        that: zbxproxy_active_proxyaddress is changed
+
+    - name: test - update Zabbix proxy server to be active use proxy_address (again)
+      zabbix_proxy:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        proxy_name: zbxproxy_example01
+        description: Example Zabbix Proxy
+        state: present
+        status: active
+        proxy_address: 10.1.1.0/24,zabbix.example.com
+      register: zbxproxy_active_proxyaddress_again
+
+    - name: assert that nothing has been changed
+      assert:
+        that: not zbxproxy_active_proxyaddress_again is changed
+
+
+    - name: test - update Zabbix proxy server to use encryption
+      zabbix_proxy:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        proxy_name: zbxproxy_example01
+        description: Example Zabbix Proxy
+        state: present
+        status: active
+        tls_psk_identity: test
+        tls_connect: PSK
+        tls_accept: PSK
+        tls_psk: 123456789abcdef123456789abcdef12
+      register: zbxproxy_encryption
+
+    - name: assert that encryption has been enabled
+      assert:
+        that: zbxproxy_encryption is changed
+
+    - name: test - update Zabbix proxy server to use encryption (again)
+      zabbix_proxy:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        proxy_name: zbxproxy_example01
+        description: Example Zabbix Proxy
+        state: present
+        status: active
+        tls_psk_identity: test
+        tls_connect: PSK
+        tls_accept: PSK
+        tls_psk: 123456789abcdef123456789abcdef12
+      register: zbxproxy_encryption_again
+
+    - name: assert that nothing has been changed
+      assert:
+        that: not zbxproxy_encryption_again is changed
+
+    - name: test - update Zabbix proxy server encryption settings
+      zabbix_proxy:
+        server_url: "{{ zabbix_server_url }}"
+        login_user: "{{ zabbix_login_user }}"
+        login_password: "{{ zabbix_login_password }}"
+        proxy_name: zbxproxy_example01
+        description: Example Zabbix Proxy
+        state: present
+        status: active
+        tls_psk_identity: test
+        tls_connect: PSK
+        tls_accept: PSK
+        tls_psk: 123456789abcdef123456789abcdef12
+        tls_issuer: AcmeCorp
+        tls_subject: AcmeCorpServer
```

Comment:
Is this a test for `tls_connect=certificate` ?(typo?)
If when `tls_connect=PSK` ignore `tls_issuer` and `tls_subject` parameters.
If you enable this parameters, `tls_connect` must be `certificate`.

Reply:
Copied from zabbix_host tests. I would say it is a blob of settings just to prove that all of them can be set and are idempotent when rerun again. 

It would be different if we wanted to test actual state of the host on zabbix server (maybe via _info modules), but so far we are not doing that. 

Will correct it to make more sense. Thanks for noticing :)

Reply:
Should be ok now @sky-joker!